File: 1-Data Preprocessing/Python/data_preprocessing_template.py
# Data Preprocessing

# ------------- Importing the libraries -------
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# ------------- Importing the dataset --------------
dataset = pd.read_csv('Data.csv')
X = dataset.iloc[:, :-1].values
Y = dataset.iloc[:, 3].values


# --------------- Splitting into Training & Test Set -----------------

from sklearn.cross_validation import train_test_split

X_train, X_test, Y_train, Y_Test = train_test_split(X, Y, # array i.e the matrix
                                                    test_size = 0.2, # 20% test data
                                                    random_state = 0) # 


# --------------- Feature Scaling ------------

"""from sklearn.preprocessing import StandardScaler

sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train) # we have to to fit the sc scaler and then transform
X_test = sc_X.transform(X_test) # as we have made the fit to training set, in test set
                                # only transform is required"""

File: 1-Data Preprocessing/R/data_preprocessing_template.R

#First you have to set the working directory from "More" option menu

# --------- Importing the dataset -------------
dataset = read.csv('Data.csv')
#dataset = dataset[, 2:3]

# --------------- Splitting into Training & Test Set -----------------

#install.packages('caTools') # we have to download this liabrary for splitting

library(caTools) # to include the liabrary to the script

set.seed(123)

#This will return only True and False
#                   True = observation in training set
#                   False = observation in test set

split = sample.split(dataset$Purchased, # put the dependant variable only
                     SplitRatio = 0.8) # Training set 80%


training_set =  subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)


#----------- Feature Scaling -----------------------

#training_set =  scale(training_set) # this will show an error as coloumn 1 and 4 
                                    # are factors not numeric
#test_set =  scale(test_set)

#training_set[, 2:3] =  scale(training_set[, 2:3])
#test_set[, 2:3] =  scale(test_set[, 2:3])



